lib/service/kubernetes.go

```diff
@@ -298,7 +296,8 @@ func (process *TeleportProcess) initKubernetesService(log *logrus.Entry, conn *C
 			warnOnErr(kubeServer.Close(), log)
 			agentPool.Stop()
 		}
-		warnOnErr(listener.Close(), log)
+
+		warnOnErr(asyncEmitter.Close(), log)
```

Comment:
Can `asyncEmitter` no longer be `nil`? The `if asyncEmitter != nil` condition is no longer here.

Reply:
Reduced PR scope.

Reply:
The same comment applies to the `asyncEmitter` in `proxy.shutdown`, which was never nullable in the first place.